mitodl/ocw-course-hugo-starter
# optional build arguments

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-course-hugo-starter/issues/4
Closes https://github.com/mitodl/ocw-course-hugo-starter/issues/3

#### What's this PR do?
This PR adds some structure to the script along with the ability to parse new options, found in the readme.

#### How should this be manually tested?
Clone and start [ocw-www](https://github.com/mitodl/ocw-www) with:

```sh
git clone git@github.com:mitodl/ocw-www.git
git fetch
git checkout cg/hugo-server-rendertodisk
yarn install
npm start
```

This will start hugo server with the `--renderToDisk` option which allows us to inject content into the running dev server.  Now, we are ready to get OCW content with [ocw-to-hugo](https://github.com/mitodl/ocw-to-hugo):

```sh
git clone git@github.com:mitodl/ocw-to-hugo.git
yarn install
node . -i private/courses -o private/output --courses course_json_examples/example_courses.json --download
```

Once the courses have downloaded and conversion has finished, we are ready to build and inject them into our running `ocw-www` site.  In your `ocw-course-hugo-starter` folder run the following, replacing `ocw-to-hugo` and `ocw-www` with your local paths to those repos:

```sh
yarn install
./build.sh -o ocw-www/site/public -c ocw-to-hugo/private/output -b //localhost:3000/courses
```

After the build is done, you should be able to browse the following course sites and any other course in `example_courses.json`:

http://localhost:3000/courses/18-06-linear-algebra-spring-2010/
http://localhost:3000/courses/12-001-introduction-to-geology-fall-2013/
